### PR TITLE
🐛 Fix another webpack 5 deprecation warning

### PR DESCRIPTION
**Pull request by RDIL**
## Summary
Just one more to go.  This one was just 2 `Array` to `Set` changes.

Not sure why the lockfile got updated, but I also fixed a typo and some warnings outputted while running the tests.

## Why
Its a warning.  

## Checklist

- [X] Your code builds clean without any `errors` or `warnings`
- [X] You are using `approved terminology`
- [X] You have added `unit tests`, if apply.

## Emojis for categorizing pull requests:

⚡️ New feature (`:zap:`)
🐛 Bug fix (`:bug:`)  
🔥 P0 fix (`:fire:`)  
✅ Tests (`:white_check_mark:`)  
🚀 Performance improvements (`:rocket:`)  
🖍 CSS / Styling (`:crayon:`)  
♿ Accessibility (`:wheelchair:`)  
🌐 Internationalization (`:globe_with_meridians:`)  
📖 Documentation (`:book:`)  
🏗 Infrastructure / Tooling / Builds / CI (`:building_construction:`)  
⏪ Reverting a previous change (`:rewind:`)  
♻️ Refactoring (like moving around code w/o any changes) (`:recycle:`)  
🚮 Deleting code (`:put_litter_in_its_place:`)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
   },
   "peerDependencies": {
     "react-loadable": "*",
-    "webpack": ">=4.41.1"
+    "webpack": ">=4.41.1 || 5.x"
```
Comment (themgoncalves) [inline package.json:47]: That is something I was missing. Thanks for addressing it.

File: source/ReactLoadableSSRAddon.js [hunk 2]
```diff
@@ -195,7 +195,9 @@ class ReactLoadableSSRAddon {
       return 0;
     };
 
-    return compilationChunks.reduce((chunks, chunk) => {
+    return (
+      WEBPACK_5 ? Array.from(compilationChunks) : compilationChunks
```
Comment (themgoncalves) [inline source/ReactLoadableSSRAddon.js]: As this _logic_ is being used on 2 places, it's a great idea to abstract it into an util, so we have code reusability.

E.g.

```js
// source/utils/toArray.js

/**
 * Ensure that given source is an array
 * @method toArray
 * @function
 * @param {*} source
 * @returns {*[]}
 */
export default function toArray(source) {
  return Array.isArray(source) ? source : [source];
}
```

And then we can finally use:

```suggestion
      toArray(compilationChunks)
```

File: source/ReactLoadableSSRAddon.js [hunk 3]
```diff
@@ -215,9 +217,10 @@ class ReactLoadableSSRAddon {
         chunks.push({
           id,
           names: chunk.name ? [chunk.name] : [],
-          files: chunk.files.slice(),
+          files: (WEBPACK_5 ? Array.from(chunk.files) : chunk.files).slice(),
```
Comment (themgoncalves) [inline source/ReactLoadableSSRAddon.js]:
```suggestion
          files: toArray(chunk.files).slice(),
```

Same thing here

File: source/utils/computeIntegrity.js [hunk 4]
```diff
@@ -9,7 +9,7 @@ import crypto from 'crypto';
 /**
  * Compute SRI Integrity
  * @func computeIntegrity
- * See {@link https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity|Subresource Integrity} at MDN
+ * See {@link https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity Subresource Integrity} at MDN
```
Comment (themgoncalves) [inline source/utils/computeIntegrity.js:12]: Also, could you update this link to `https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity`, as it's currently broken.

**Review by themgoncalves — COMMENTED** [review]
Thanks for the PR!

File: source/ReactLoadableSSRAddon.js (hunk 2 above)
Comment (RDIL) [reply source/ReactLoadableSSRAddon.js]: This is a `Set` to `Array` change, so `[source] doesn't work.  `

Comment (themgoncalves) [reply source/ReactLoadableSSRAddon.js]: Not a problem!

File: source/utils/computeIntegrity.js (hunk 4 above)
Comment (RDIL) [reply source/utils/computeIntegrity.js:12]: The updated link works just fine for me:
Capture

File: webpack.config.js [hunk 5]
```diff
@@ -6,11 +6,12 @@ const HOST = process.env.HOST || '127.0.0.1';
 const PORT = process.env.PORT || '8080';
 
 module.exports = {
+  mode: 'production',
   target: 'web',
   entry: {
     index: './example/client.jsx',
   },
-  devtool: 'cheap-module-eval-source-map',
+  devtool: 'eval-cheap-module-source-map',
```
Comment (RDIL) [inline webpack.config.js:14]: For testing against webpack 5 to work, this change is required, and the other 2 fix warnings.

File: source/ReactLoadableSSRAddon.js (hunk 3 above)
Comment (RDIL) [reply source/ReactLoadableSSRAddon.js]: Fixed!

**Review by themgoncalves — APPROVED** [review]
LGTM

**Pull request merged**